File: app/src/main/java/com/letianpai/robot/control/mode/callback/ModeChangeCallback.java
package com.letianpai.robot.control.mode.callback;

/**
 * 模式切换状态回调
 * @author liujunbin
 */
public class ModeChangeCallback {

    private ModeChangeListener mModeChangeListener;

    private static class MModeChangeCallbackHolder {
        private static ModeChangeCallback instance = new ModeChangeCallback();
    }

    private ModeChangeCallback() {

    }

    public static ModeChangeCallback getInstance() {
        return MModeChangeCallbackHolder.instance;
    }

    public interface ModeChangeListener {
        void onViewModeChanged(int viewMode,int modeStatus);
    }

    public void setViewModeChangeListener(ModeChangeListener listener) {
        this.mModeChangeListener = listener;
    }

    public void setModeChange(int viewMode,int modeStatus) {
        if (mModeChangeListener != null) {
            mModeChangeListener.onViewModeChanged(viewMode,modeStatus);
        }
    }
}

File: app/src/main/java/com/letianpai/robot/control/mode/callback/ModeChangeCallback.kt
package com.letianpai.robot.control.mode.callback

/**
 * 模式切换状态回调
 * @author liujunbin
 */
class ModeChangeCallback private constructor() {
    private var mModeChangeListener: ModeChangeListener? = null

    private object MModeChangeCallbackHolder {
        val instance: ModeChangeCallback = ModeChangeCallback()
    }

    interface ModeChangeListener {
        fun onViewModeChanged(viewMode: Int, modeStatus: Int)
    }

    fun setViewModeChangeListener(listener: ModeChangeListener?) {
        this.mModeChangeListener = listener
    }

    fun setModeChange(viewMode: Int, modeStatus: Int) {
        if (mModeChangeListener != null) {
            mModeChangeListener!!.onViewModeChanged(viewMode, modeStatus)
        }
    }

    companion object {
        val instance: ModeChangeCallback
            get() {
                return MModeChangeCallbackHolder.instance
            }
    }
}
